astrolabe/spec_runner.py

```diff
@@ -213,33 +210,40 @@ def run(self, persist_cluster=False):
         # Step-5: interrupt driver workload and capture streams
         LOGGER.info("Stopping workload executor [PID: {}]".format(
             worker_subprocess.pid))
-        os.kill(worker_subprocess.pid, self.sigint)
-        stdout, stderr = worker_subprocess.communicate()
-        LOGGER.info("Stopped workload executor")
+        os.killpg(worker_subprocess.pid, self.sigint)
+        stdout, stderr = worker_subprocess.communicate(timeout=10)
+        LOGGER.info("Stopped workload executor [exit code: {}]".format(
+            worker_subprocess.returncode))
 
         # Stop the timer
         timer.stop()
 
         # Step-6: compute xunit entry.
         junit_test = junitparser.TestCase(self.id)
         junit_test.time = timer.elapsed
-        if worker_subprocess.returncode != 0:
+
+        try:
+            err_info = json.loads(stderr)
+        except json.JSONDecodeError:
+            err_info = {'numErrors': '-1', 'numFailures': '-1'}
```

Comment:
Can we change these to be numbers (`'-1'` to `-1`) so that we don't need to call `int` later?

Reply:
What about the case when we parse output returned by the workload executor? I was under the impression that the numbers in the JSON returned by that script would also end up getting parsed as string so I needed to perform this type-conversion regardless.

Reply:
`json.loads` does the number parsing:
```
>>> import json
>>> json.loads('{"num": -1}')
{'num': -1}
```